### ISSUE TITLE
Default sort

### ISSUE THREAD

**Issue by lossoth** [alfajango/jquery-dynatable#91]
Is it possible to set a default sort using data-dynatable-column?

**Comment by piwi91** [comment]
This doesn't work for me either. The problem is in the mergeSettings, line 159.

```javascript
var newOptions = $.extend(true, {}, defaults, options);
```

The options object does have the `sort` configuration but after the merge it isn't in the `newOptions` object.

**Comment by hugoduraes** [comment]
Yep, settings are not being merged, as @piwi91 said.

**Comment by umdstu** [comment]
Any updates on addresses this deep copy issue in the mergeSettings function? I even tried changing the queries/sorts default hashes in the file itselt, and they still don't survive the deep copy. 

Also, this is effecting default queries, not just sorts, AFAIK.

**Comment by DraZen10** [comment]
Using the example in [JSFiddle](http://jsfiddle.net/kgonkxyn/3) posted by @ksireeshacould I saw sort function working although it does not in my implementation.
Found a workaround just adding the variable <b>sort[<i>field_for_sorting</i>]=-1</b> into the http_get vars as I open the table from a html form.
Hope this helps someone

**Comment by umdstu** [comment]
That was my solution as well, DraZen10. Clearly not at all ideal, but it
has been working so far.

**Comment by rstuart85** [comment]
For anyone that is wondering, this issue appears to be fixed in the master branch.

**Comment by DraZen10** [comment]
Yes! It works. I see no change in version or revision, but the file is slightly different and the issue seems to be fixed. Thanks!

**Comment by umdstu** [comment]
I don't understand. The commit history shows no recent commits. Where was the change?

**Comment by rstuart85** [comment]
Not sure when the change was committed @umdstu but the problem was in the [init method of the Sorts object](https://github.com/alfajango/jquery-dynatable/blob/master/jquery.dynatable.js#L907). It would always set `settings.dataset.sorts` to null if there were no GET params. It's stopped doing that now.

**Comment by rstuart85** [comment]
Looks like it was fixed over a year ago in b7ead6a1fb61aadbbf3b63531b828b823faa70ee.

**Comment by aklatzke** [comment]
If anyone else is having this issue and doesn't have the luxury to wait until that commit (b7ead6a) is merged in, the changes are fairly easy to merge in manually. In the stable version of the dynatable script:

Change line 103:

`sortKeys: null` -> `sortKeys : []`

Line 193, change:

`if (!this.settings.dataset.ajax || (this.settings.dataset.ajax && this.settings.dataset.ajaxOnLoad) || this.settings.features.paginate)`

to 

`if (!this.settings.dataset.ajax || (this.settings.dataset.ajax && this.settings.dataset.ajaxOnLoad) || this.settings.features.paginate || (this.settings.features.sort && !$.isEmptyObject(this.settings.dataset.sorts)))`

Replace the Sorts.init function at line 871 with the following:

    this.init = function() {
      var sortsUrl = window.location.search.match(new RegExp(settings.params.sorts + '[^&=]*=[^&]*', 'g'));
      if (sortsUrl) {
        settings.dataset.sorts = utility.deserialize(sortsUrl)[settings.params.sorts];
      }
      if (!settings.dataset.sortsKeys.length) {
        settings.dataset.sortsKeys = utility.keysFromObject(settings.dataset.sorts);
      }
    };

Hope this helps someone.

**Comment by Silver89** [comment]
Doesn't look like `data-dynatable-sorts` got acted upon? Is there a way I can sort any table on the first column by default with the same code? Would be really useful and stop me having to write a custom function... unless there is one that does this?

**Comment by yogi1135** [comment]
Guys you are using 'sorts' key on the param, you should be using it on the input field 
inputs:{
               {date_missing: -1},
    },

**Comment by JangoSteve** [comment]
I haven't had a chance to figure out why this isn't working. The reason is exactly as @piwi91 described in [this comment](https://github.com/alfajango/jquery-dynatable/issues/91#issuecomment-70254879). I have no idea why the jquery `extend` function causes the sorts object to be deleted. I don't have time to debug it at the moment, but an alternative way to get default sorting working in case it helps anyone is to bind to the `dynatable:init` event and add the default sort programmatically on initialization, like this:

```js
    $('#my-table')
      .bind('dynatable:init', function(e, dynatable) {
        dynatable.sorts.add('sort_column', -1);
      })
      .dynatable();
```

**Comment by msaus** [comment]
I guess this is not still fixed.
